src/xoauth2.js

```diff
@@ -161,7 +169,8 @@ XOAuth2Generator.prototype.generateToken = function(callback) {
         }
 
         if (data.access_token) {
-            self.updateToken(data.access_token, data.expires_in);
+            self.updateToken(data.access_token, data.expires_in, data.id_token);
+            // It's unnecessary to add the id_token to the callback - all we need to do is emit it for later use.
```

Comment:
Curious do you have any concerns with providing it to the callback? It might prove useful to clients.

Reply:
I just wanted to limit the scope of my changes given how we use this library w.r.t. sending - also since I'd forked this for our internal usage (where we only need the id token to store later), it struck me as somewhat spurious to make that change.